corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -0,0 +1,117 @@
+                container: 'geospatial-map', // container ID
+                style: 'mapbox://styles/mapbox/streets-v12', // style URL
+                center: centerCoordinates, // starting position [lng, lat]
+                zoom: 12,
+                attribution: '© <a href="https://www.mapbox.com/about/maps/">Mapbox</a> ©' +
+                             ' <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+            });
+
+            const draw = new MapboxDraw({
+                // API: https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md
+                displayControlsDefault: false,
+                boxSelect: true, // enables box selection
+                controls: {
+                    polygon: true,
+                    trash: true
+                },
+            });
+            map.addControl(draw);
+
+            map.on('draw.create', updateArea);
+            map.on('draw.delete', updateArea);
+            map.on('draw.update', updateArea);
+
+            function getCoordinates(event) {
+                return event.lngLat;
+            };
+
+            function updateArea(e) {
+                const data = draw.getAll();
+                const area = turf.area(data);
+                // Restrict the area to 2 decimal points.
+                const rounded_area = Math.round(area * 100) / 100;
+            }
+
+            // We should consider refactoring and splitting the below out to a new JS file
+            let clickedMarker;
+
+            self.addCaseMarkersToMap = function (cases) {
+                const markerColor = "#00FF00";
+                cases.forEach(element => {
+                    let coordinates = element.coordinates;
+                    if (coordinates && coordinates.lat && coordinates.lng) {
+                        self.addMarker(coordinates, markerColor);
+                    }
+                });
+            };
+
+            self.addMarker = function (coordinates, color) {
+                const marker = new mapboxgl.Marker({color: color, draggable: false});
+                marker.setLngLat(coordinates);
+                marker.addTo(map);
+            }
+
+            function moveMarkerToClickedCoordinate(coordinates) {
+                if (clickedMarker != null) {
+                    clickedMarker.remove();
+                }
+                if (draw.getMode() === 'draw_polygon') {
+                    // It's weird moving the marker around with the ploygon
+                    return;
+                }
+                clickedMarker = new mapboxgl.Marker({color: "FF0000", draggable: true});
+                clickedMarker.setLngLat(coordinates);
+                clickedMarker.addTo(map);
+            }
+
+            function showCreateCasePopup(coordinates) {
+                new mapboxgl.Popup()
+                .setLngLat(coordinates)
+                .setHTML(coordinates + '<br/><button> Create Case at this point </button>')
```

Comment:
Would it be possible to do this using KnockoutJS?

Reply:
I'm not sure how we would do that here?

Reply:
I think I picked a wrong line, but generally I was suggesting to use KnockoutJS observables where we need to propagate or update changes to make the code manageable. 

But given that the PR is already quite challenging we can consider that in a latter PR.

Reply:
don't see a screen shot including this button, but my guess is it should use `btn btn-default` classes

Reply:
Apologies, this stuff is actually not used at the moment. I included this code for when we need to implement creating cases (it's coming), but I'll remove it rather. Removed: [5fab144](https://github.com/dimagi/commcare-hq/pull/33020/commits/5fab1448824f511b5a817e35414cfdbe2cd00c9a)

Reply:
cool, thanks